masonry_winit/src/event_loop_runner.rs

```diff
@@ -458,6 +457,58 @@ impl MasonryState<'_> {
         window.handle.set_ime_allowed(false);
     }
 
+    // --- MARK: REDRAW
+    fn redraw(&mut self, handle_id: HandleId) {
+        let _span = info_span!("redraw");
+
+        let window = self.windows.get_mut(&handle_id).unwrap();
+        let size = window.render_root.size();
+        if size.width == 0 || size.height == 0 {
+            // Surface can't have a dimension of zero, remove the stale surface to save memory.
+            self.surfaces.remove(&handle_id);
+            return;
+        }
+
+        // Get the existing surface or create a new one
+        let surface = if let Some(surface) = self.surfaces.get_mut(&handle_id) {
+            // The window might have been resized, make sure the surface dimensions match.
+            if surface.config.width != size.width || surface.config.height != size.height {
+                self.render_cx
+                    .resize_surface(surface, size.width, size.height);
+            }
+            surface
+        } else {
+            let surface = create_surface(&mut self.render_cx, window.handle.clone(), size);
+            self.surfaces.insert(handle_id, surface);
+            self.surfaces.get_mut(&handle_id).unwrap()
+        };
+
+        let now = Instant::now();
+        // TODO: this calculation uses wall-clock time of the paint call, which
+        // potentially has jitter.
+        //
+        // See https://github.com/linebender/druid/issues/85 for discussion.
+        let last = self.last_anim.take();
+        let elapsed = last.map(|t| now.duration_since(t)).unwrap_or_default();
+
+        window
+            .render_root
+            .handle_window_event(WindowEvent::AnimFrame(elapsed));
```

Comment:
Fwiw, we probably need to handle signals after an animation (see e.g. #985). That being said, everything about animations will need to change in the future anyway, and this isn't a regression here.

Reply:
Makes sense, but yeah it wasn't done before either, so that'll be future work.